peer/x/roundrobin/list.go

```diff
@@ -168,7 +164,22 @@ func (pl *List) addToAvailablePeers(p peer.Peer) error {
 
 // Start notifies the List that requests will start coming
 func (pl *List) Start() error {
-	return pl.once.Start(nil)
+	return pl.once.Start(pl.start)
+}
+
+func (pl *List) start() error {
+	pl.lock.Lock()
+	defer pl.lock.Unlock()
+
+	var errs error
+	for _, pid := range pl.uninitializedPeers {
+		errs = multierr.Append(errs, pl.addPeerIdentifier(pid))
+		delete(pl.uninitializedPeers, pid.Identifier())
+	}
+
+	pl.shouldRetainPeers.Store(true)
+
+	return errs
```

Comment:
Should `shouldRetainPeers` be changed even when `start()` failed? IIRC, LifecycleOnce will consider the object not started if start fails.